utils/cautils/token_generator.go

```diff
@@ -193,6 +195,54 @@ func generateX5CToken(ctx *cli.Context, p *provisioner.X5C, tokType int, tokAttr
 	}
 }
 
+func generateNebulaToken(ctx *cli.Context, p *provisioner.Nebula, tokType int, tokAttrs tokenAttrs) (string, error) {
+	certFile := ctx.String("nebula-cert")
+	keyFile := ctx.String("nebula-key")
+	if certFile == "" {
+		return "", errs.RequiredWithProvisionerTypeFlag(ctx, "Nebula", "nebula-cert")
+	}
+	if keyFile == "" {
+		return "", errs.RequiredWithProvisionerTypeFlag(ctx, "Nebula", "nebula-key")
+	}
+
+	// Get private key from given key file, nebula CAs uses ed25519 keys while
+	// nebula leafs uses X25519 keys.
+	jwk, err := jose.ReadKey(keyFile)
+	if err != nil {
+		return "", err
+	}
+
+	var key []byte
+	switch k := jwk.Key.(type) {
+	case x25519.PrivateKey:
```

Comment:
you can combine these cases into one line, since the actions is the same

Reply:
You cannot because they are different types, even if the underlying type is the same.